src/http_module.cpp

```diff
@@ -702,6 +715,45 @@ char* addResourceAttr(ngx_conf_t* cf, ngx_command_t* cmd, void* conf)
     return NGX_CONF_OK;
 }
 
+char* setTrustedCertificate(ngx_conf_t* cf, ngx_command_t* cmd, void* conf) {
+    auto ret = ngx_conf_set_str_slot(cf, cmd, conf);
+    if (ret != NGX_CONF_OK) {
+        return ret;
+    }
+
+    auto mcf = getMainConf(cf);
+
+    if (mcf->trustedCertificate.len > 0) {
+        if (ngx_get_full_name(cf->pool, &cf->cycle->prefix,
+                &mcf->trustedCertificate) != NGX_OK) {
+            ngx_conf_log_error(NGX_LOG_EMERG, cf, 0,
+                "Failed to get full path for \"%V\"", mcf->trustedCertificate);
+            return (char*)NGX_CONF_ERROR;
+        }
+
+        std::ifstream file{{(const char *)mcf->trustedCertificate.data,
```

Comment:
This still needs to be under `try`, as fstreams's constructor is not `noexcept`.